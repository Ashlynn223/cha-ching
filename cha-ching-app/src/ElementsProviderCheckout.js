import { Elements } from "@stripe/react-stripe-js";
import { loadStripe } from "@stripe/stripe-js";
import CheckoutForm from "./CheckoutForm";
import Error from "./Error";
import Loading from "./Loading";
import { useLocation } from "react-router-dom";

const stripeElementStyling = {
  variables: {
    fontFamily: "Sohne, system-ui, sans-serif",
    fontSizeBase: "20px",
    fontWeightNormal: "500",
    borderRadius: "8px",
    colorPrimary: "white",
    colorText: "#8f424b",
    colorTextSecondary: "#8f424b",
    colorTextPlaceholder: "white",
    colorIconTab: "white",
    colorIconTabSelected: "#1A1B25",
    colorLogo: "dark",
  },
  rules: {
    ".Tab": {
      backgroundColor: "transparent",
    },
    ".Tab--selected": {
      backgroundColor: "#FFCE48",
      color: "#1A1B25",
    },
    ".Input": {
      backgroundColor: "transparent",
      border: "1.5px solid #8f424b",
    },
  },
};

const stripePromise = loadStripe(process.env.REACT_APP_PUBLIC_STRIPE_KEY);
const ElementsProviderCheckout = () => {
  const isEmpty = (object) => Object.keys(object).length === 0;
  const { state: options } = useLocation();
  const styledOptions = {clientSecret: options.clientSecret, stripeElementStyling}
  console.log("styledOptions", styledOptions)

  return !isEmpty(options) ? (
    options.clientSecret === null ? (
      <Error />
    ) : (
      <Elements options={styledOptions} stripe={stripePromise}>
        <CheckoutForm />
      </Elements>
    )
  ) : (
    <Loading />
  );
};

export default ElementsProviderCheckout;
