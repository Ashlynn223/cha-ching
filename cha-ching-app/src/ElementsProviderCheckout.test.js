import { render, screen } from "@testing-library/react";
import ElementsProviderCheckout from "./ElementsProviderCheckout";
import { MemoryRouter } from "react-router-dom";

describe("<ElementsProviderCheckout/>", () => {
  test("renders Loading when options are undefined", async () => {
    const component = render(
      <MemoryRouter initialEntries={[{ pathname: "/checkout", state: {} }]}>
        <ElementsProviderCheckout />
      </MemoryRouter>,
    );
    expect(await screen.findByRole("alert")).toBeInTheDocument();
    expect(component.asFragment()).toMatchSnapshot();
  });

  test("renders Error when clientSecret is null", async () => {
    const component = render(
      <MemoryRouter
        initialEntries={[
          { pathname: "/checkout", state: { clientSecret: null } },
        ]}
      >
        <ElementsProviderCheckout />
      </MemoryRouter>,
    );
    expect(await screen.findByText("error")).toBeInTheDocument();
    expect(component.asFragment()).toMatchSnapshot();
  });

  test("renders payment form when clientSecret is present", async () => {
    const component = render(
      <MemoryRouter
        initialEntries={[
          { pathname: "/checkout", state: { clientSecret: "my_fake_secret" } },
        ]}
      >
        <ElementsProviderCheckout />
      </MemoryRouter>,
    );
    expect(screen.queryByRole("alert")).not.toBeInTheDocument();
    expect(screen.queryByText("error")).not.toBeInTheDocument();
    expect(await screen.findByRole("form")).toBeInTheDocument();
    expect(component.asFragment()).toMatchSnapshot();
  });
});
