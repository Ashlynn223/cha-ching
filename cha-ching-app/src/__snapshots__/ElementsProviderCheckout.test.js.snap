// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ElementsProviderCheckout/> renders Error when clientSecret is null 1`] = `
<DocumentFragment>
  <div
    class="errorWrapper"
  >
    <div
      class="errorContainer"
    >
      <div
        class="errorText"
      >
        error
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ElementsProviderCheckout/> renders Loading when options are undefined 1`] = `
<DocumentFragment>
  <div
    class="spinnerWrapper"
  >
    <div
      aria-live="polite"
      class="spinnerContainer"
      role="alert"
    />
  </div>
</DocumentFragment>
`;

exports[`<ElementsProviderCheckout/> renders Stripe's Elements provider with clientSecret 1`] = `
<DocumentFragment>
  <div>
    <div
      class="header"
    >
      Cha-ching
    </div>
    <div
      class="paymentFormContainer"
    >
      <form
        name="payment-form"
      >
        <div
          class="emailLabel"
        >
          Email
        </div>
        <input
          class="emailInput"
        />
        <div>
          <div />
        </div>
      </form>
      <button
        class="button"
      >
        Submit
      </button>
    </div>
  </div>
</DocumentFragment>
`;
